pkg/cmd/openshift-sdn-node/sdn.go

```diff
@@ -1,23 +1,23 @@
 package openshift_sdn_node
 
 import (
+	"fmt"
 	"io/ioutil"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/kubernetes/scheme"
-	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
-	"k8s.io/client-go/tools/record"
+	eventsv1 "k8s.io/client-go/tools/events"
 
 	sdnnode "github.com/openshift/sdn/pkg/network/node"
 )
 
 const openshiftCNIFile string = "/etc/cni/net.d/80-openshift-network.conf"
 
 // initSDN sets up the sdn process.
-func (sdn *openShiftSDN) initSDN() error {
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartRecordingToSink(&corev1client.EventSinkImpl{Interface: sdn.informers.kubeClient.CoreV1().Events("")})
-	sdn.sdnRecorder = eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "openshift-sdn", Host: sdn.nodeName})
+func (sdn *openShiftSDN) initSDN(stopCh <-chan struct{}) error {
+	eventBroadcaster := eventsv1.NewBroadcaster(&eventsv1.EventSinkImpl{Interface: sdn.informers.kubeClient.EventsV1()})
+	eventBroadcaster.StartRecordingToSink(stopCh)
+	sdn.sdnRecorder = eventBroadcaster.NewRecorder(scheme.Scheme, fmt.Sprintf("openshift-sdn-node-%s", sdn.nodeName))
```

Comment:
It appears that the new API has both `ReportingController` and `ReportingInstance`, where the string passed here becomes `ReportingController`, and the client-go code does:

```
	hostname, _ := os.Hostname()
	reportingInstance := reportingController + "-" + hostname
```

so we shouldn't manually add the nodename. Just keep passing "openshift-sdn"